### ISSUE TITLE
Use php_uname() instead of PHP_OS

### ISSUE THREAD

**Issue by derUli** [tivie/php-os-detector#3]
Please note that PHP_OS returns the Operating System where the PHP executable was built.
While in most cases this is equally to the runtime OS, this returns an invalid value if the php interpreter was cross built on another Operating System.

http://php.net/manual/de/function.php-uname.php

**Comment by drupol** [comment]
This makes perfect sense.

**Comment by drupol** [comment]
I just submitted the pull request, a review would be welcome.

**Comment by drupol** [comment]
I ended up doing my own package with a complete different logic: https://github.com/drupol/phposinfo

It's also using php_uname().

**Comment by WyriHaximus** [comment]
@derUli out of interest do you have any examples where you can into this?

**Comment by danepowell** [comment]
I agree this makes sense. It's probably rare that people cross compile PHP, but I'm sure it happens. It's simply not right to be using PHP_OS when a more accurate and reliable alternative (php_uname) exists. The PHP documentation for each of those members makes it pretty clear that this is best practice.

**Comment by drupol** [comment]
@WyriHaximus I don't have any example under the hand, but I guess that some platforms (RaspBerry, etc etc) are using cross compiled binaries.

@danepowell :+1:

**Comment by tivie** [comment]
does anyone have a list of possible `php_uname('s')` values? Because they don't match PHP_OS
